src/services/JobRoleAPIService.ts

```diff
@@ -0,0 +1,34 @@
+import axios from "axios";
+import type { JobRole } from "../models/job-role.js";
+import type { JobRoleservice } from "./interfaces.js";
+
+export class JobRoleAPIService implements JobRoleservice {
+  async getAllJobs(): Promise<JobRole[]> {
+    const jobs: JobRole[] = [];
+    const response = await axios.get<JobRole[]>(
+      "http://localhost:3001/api/jobs"
+    );
+    jobs.push(...response.data);
+    return jobs;
+  }
+
+  async getJobById(id: number): Promise<JobRole | undefined> {
+    try {
+      const response = await axios.get<JobRole>(
+        `http://localhost:3001/api/jobs/${id}`
+      );
+      return response.data;
+    } catch (_error) {
+      return undefined;
```

Comment:
it would be good to log the error here. you can just use `console.log` for now.
This will make debugging easier.